src/include/duckdb/planner/binder.hpp

```diff
@@ -75,6 +76,8 @@ class Binder {
 	bool requires_valid_transaction = true;
 	//! The alias for the currently processing subquery, if it exists
 	string alias;
+	//! Macro parameter bindings (if any)
+	shared_ptr<MacroBinding> macro_binding = nullptr;
```

Comment:
Does this need to be a shared pointer? We prefer to avoid shared pointers unless absolutely required (because of the performance penalty, potential leaks because of referential cycles, difficult to reason about ownership). Could this be turned into a unique pointer, or a regular (non-smart) pointer?

Reply:
In my first idea of macro bindings, this was needed. However, I've changed the approach since then, and it is no longer needed. A regular pointer will suffice.